from PyQt5 import QtCore, QtGui, QtWidgets
from PyQt5.QtWidgets import QMessageBox
from herramientas.controlLogin import *
from herramientas.controlMain import controlMain

class Ui_MainWindow(object):
    def setupUi(self, MainWindow):
        MainWindow.setObjectName("MainWindow")
        MainWindow.resize(939, 601)
        MainWindow.setStyleSheet("background-color: rgb(135, 135, 135);")
        self.centralwidget = QtWidgets.QWidget(MainWindow)
        self.centralwidget.setObjectName("centralwidget")
        self.tablaVenci = QtWidgets.QTableWidget(self.centralwidget)
        self.tablaVenci.setEnabled(False)
        self.tablaVenci.setGeometry(QtCore.QRect(20, 70, 891, 341))
        font = QtGui.QFont()
        font.setFamily("Arial")
        font.setPointSize(11)
        font.setBold(False)
        font.setWeight(50)
        self.tablaVenci.setFont(font)
        self.tablaVenci.setStyleSheet("background-color: rgb(255, 255, 255);")
        self.tablaVenci.setSelectionMode(QtWidgets.QAbstractItemView.SingleSelection)
        self.tablaVenci.setSelectionBehavior(QtWidgets.QAbstractItemView.SelectRows)
        self.tablaVenci.setObjectName("tablaVenci")
        self.tablaVenci.setColumnCount(5)
        self.tablaVenci.setRowCount(0)
        item = QtWidgets.QTableWidgetItem()
        font = QtGui.QFont()
        font.setFamily("Verdana")
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        item.setFont(font)
        self.tablaVenci.setHorizontalHeaderItem(0, item)
        item = QtWidgets.QTableWidgetItem()
        font = QtGui.QFont()
        font.setFamily("Verdana")
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        item.setFont(font)
        self.tablaVenci.setHorizontalHeaderItem(1, item)
        item = QtWidgets.QTableWidgetItem()
        font = QtGui.QFont()
        font.setFamily("Verdana")
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        item.setFont(font)
        self.tablaVenci.setHorizontalHeaderItem(2, item)
        item = QtWidgets.QTableWidgetItem()
        font = QtGui.QFont()
        font.setFamily("Verdana")
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        item.setFont(font)
        self.tablaVenci.setHorizontalHeaderItem(3, item)
        item = QtWidgets.QTableWidgetItem()
        font = QtGui.QFont()
        font.setFamily("Verdana")
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        item.setFont(font)
        self.tablaVenci.setHorizontalHeaderItem(4, item)
        self.btnAnadir = QtWidgets.QPushButton(self.centralwidget)
        self.btnAnadir.setGeometry(QtCore.QRect(740, 470, 161, 51))
        font = QtGui.QFont()
        font.setFamily("Verdana")
        font.setPointSize(16)
        font.setBold(True)
        font.setWeight(75)
        self.btnAnadir.setFont(font)
        self.btnAnadir.setStyleSheet("color: rgb(255, 255, 255);\n"
"background-color: rgb(0, 0, 0);\n"
"border-radius: 15px;")
        self.btnAnadir.setObjectName("btnAnadir")
        self.frame = QtWidgets.QFrame(self.centralwidget)
        self.frame.setGeometry(QtCore.QRect(20, 70, 891, 501))
        self.frame.setStyleSheet("background-color: rgb(255, 255, 255);\n"
"\n"
"")
        self.frame.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.frame.setFrameShadow(QtWidgets.QFrame.Raised)
        self.frame.setObjectName("frame")
        self.txtNroU = QtWidgets.QLineEdit(self.frame)
        self.txtNroU.setGeometry(QtCore.QRect(400, 40, 31, 31))
        self.txtNroU.setObjectName("txtNroU")
        self.label = QtWidgets.QLabel(self.frame)
        self.label.setGeometry(QtCore.QRect(190, 10, 81, 21))
        font = QtGui.QFont()
        font.setFamily("Verdana")
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.label.setFont(font)
        self.label.setObjectName("label")
        self.txtDias = QtWidgets.QLineEdit(self.frame)
        self.txtDias.setGeometry(QtCore.QRect(120, 180, 211, 31))
        font = QtGui.QFont()
        font.setFamily("Arial")
        font.setPointSize(16)
        font.setBold(False)
        font.setWeight(50)
        self.txtDias.setFont(font)
        self.txtDias.setObjectName("txtDias")
        self.txtObs = QtWidgets.QTextEdit(self.frame)
        self.txtObs.setGeometry(QtCore.QRect(470, 110, 381, 91))
        font = QtGui.QFont()
        font.setFamily("Arial")
        font.setPointSize(12)
        self.txtObs.setFont(font)
        self.txtObs.setObjectName("txtObs")
        self.txtNroT = QtWidgets.QLineEdit(self.frame)
        self.txtNroT.setGeometry(QtCore.QRect(470, 40, 111, 31))
        font = QtGui.QFont()
        font.setFamily("Arial")
        font.setPointSize(16)
        font.setBold(False)
        font.setWeight(50)
        self.txtNroT.setFont(font)
        self.txtNroT.setObjectName("txtNroT")
        self.txtDetalle = QtWidgets.QTextEdit(self.frame)
        self.txtDetalle.setGeometry(QtCore.QRect(470, 240, 381, 101))
        font = QtGui.QFont()
        font.setFamily("Arial")
        font.setPointSize(12)
        self.txtDetalle.setFont(font)
        self.txtDetalle.setObjectName("txtDetalle")
        self.dateCarga = QtWidgets.QDateEdit(self.frame)
        self.dateCarga.setGeometry(QtCore.QRect(120, 110, 211, 31))
        font = QtGui.QFont()
        font.setFamily("Arial")
        font.setPointSize(11)
        self.dateCarga.setFont(font)
        self.dateCarga.setObjectName("dateCarga")
        self.txtCerrado = QtWidgets.QLineEdit(self.frame)
        self.txtCerrado.setGeometry(QtCore.QRect(830, 40, 31, 31))
        self.txtCerrado.setObjectName("txtCerrado")
        self.btnGuardar = QtWidgets.QPushButton(self.frame)
        self.btnGuardar.setGeometry(QtCore.QRect(470, 390, 161, 51))
        font = QtGui.QFont()
        font.setFamily("Arial")
        font.setPointSize(16)
        font.setBold(True)
        font.setWeight(75)
        self.btnGuardar.setFont(font)
        self.btnGuardar.setStyleSheet("color: rgb(255, 255, 255);\n"
"background-color: rgb(0, 0, 0);\n"
"border-radius: 15px;\n"
"")
        icon = QtGui.QIcon()
        icon.addPixmap(QtGui.QPixmap("img/salvar.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.btnGuardar.setIcon(icon)
        self.btnGuardar.setObjectName("btnGuardar")
        self.btnCancelar = QtWidgets.QPushButton(self.frame)
        self.btnCancelar.setGeometry(QtCore.QRect(790, 460, 91, 31))
        font = QtGui.QFont()
        font.setFamily("Arial")
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.btnCancelar.setFont(font)
        self.btnCancelar.setStyleSheet("color: rgb(0, 0, 0);\n"
"background-color: rgb(255, 0, 0);\n"
"border-radius: 15px;\n"
"")
        self.btnCancelar.setObjectName("btnCancelar")
        self.pushButton = QtWidgets.QPushButton(self.frame)
        self.pushButton.setGeometry(QtCore.QRect(20, 40, 91, 31))
        font = QtGui.QFont()
        font.setFamily("Arial")
        font.setPointSize(11)
        font.setBold(True)
        font.setWeight(75)
        self.pushButton.setFont(font)
        self.pushButton.setStyleSheet("background-color: rgb(0, 0, 0);\n"
"color: rgb(255, 255, 255);\n"
"border-radius: 10px;")
        icon1 = QtGui.QIcon()
        icon1.addPixmap(QtGui.QPixmap("img/lupaa.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.pushButton.setIcon(icon1)
        self.pushButton.setFlat(False)
        self.pushButton.setObjectName("pushButton")
        self.txtUser = QtWidgets.QLineEdit(self.frame)
        self.txtUser.setGeometry(QtCore.QRect(120, 40, 271, 31))
        font = QtGui.QFont()
        font.setFamily("Arial")
        font.setPointSize(11)
        self.txtUser.setFont(font)
        self.txtUser.setStyleSheet("")
        self.txtUser.setObjectName("txtUser")
        self.label_10 = QtWidgets.QLabel(self.frame)
        self.label_10.setGeometry(QtCore.QRect(160, 80, 141, 21))
        font = QtGui.QFont()
        font.setFamily("Verdana")
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.label_10.setFont(font)
        self.label_10.setObjectName("label_10")
        self.label_2 = QtWidgets.QLabel(self.frame)
        self.label_2.setGeometry(QtCore.QRect(150, 150, 161, 21))
        font = QtGui.QFont()
        font.setFamily("Verdana")
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.label_2.setFont(font)
        self.label_2.setObjectName("label_2")
        self.dateVencimiento = QtWidgets.QCalendarWidget(self.frame)
        self.dateVencimiento.setGeometry(QtCore.QRect(80, 250, 312, 191))
        self.dateVencimiento.setAccessibleName("")
        self.dateVencimiento.setStyleSheet("background-color: rgb(170, 170, 255);\n"
"color: rgb(0, 0, 0);")
        self.dateVencimiento.setGridVisible(True)
        self.dateVencimiento.setHorizontalHeaderFormat(QtWidgets.QCalendarWidget.ShortDayNames)
        self.dateVencimiento.setVerticalHeaderFormat(QtWidgets.QCalendarWidget.NoVerticalHeader)
        self.dateVencimiento.setDateEditEnabled(True)
        self.dateVencimiento.setObjectName("dateVencimiento")
        self.label_3 = QtWidgets.QLabel(self.frame)
        self.label_3.setGeometry(QtCore.QRect(180, 220, 111, 21))
        font = QtGui.QFont()
        font.setFamily("Verdana")
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.label_3.setFont(font)
        self.label_3.setObjectName("label_3")
        self.label_6 = QtWidgets.QLabel(self.frame)
        self.label_6.setGeometry(QtCore.QRect(470, 10, 111, 21))
        font = QtGui.QFont()
        font.setFamily("Verdana")
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.label_6.setFont(font)
        self.label_6.setObjectName("label_6")
        self.checkBox = QtWidgets.QCheckBox(self.frame)
        self.checkBox.setGeometry(QtCore.QRect(640, 40, 181, 31))
        font = QtGui.QFont()
        font.setFamily("Arial")
        font.setPointSize(12)
        self.checkBox.setFont(font)
        self.checkBox.setObjectName("checkBox")
        self.label_7 = QtWidgets.QLabel(self.frame)
        self.label_7.setGeometry(QtCore.QRect(470, 80, 111, 21))
        font = QtGui.QFont()
        font.setFamily("Verdana")
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.label_7.setFont(font)
        self.label_7.setObjectName("label_7")
        self.label_11 = QtWidgets.QLabel(self.frame)
        self.label_11.setGeometry(QtCore.QRect(470, 210, 71, 21))
        font = QtGui.QFont()
        font.setFamily("Verdana")
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.label_11.setFont(font)
        self.label_11.setObjectName("label_11")
        self.frame2 = QtWidgets.QFrame(self.frame)
        self.frame2.setGeometry(QtCore.QRect(160, 90, 561, 231))
        self.frame2.setStyleSheet("background-color: rgb(85, 170, 127);")
        self.frame2.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.frame2.setFrameShadow(QtWidgets.QFrame.Raised)
        self.frame2.setObjectName("frame2")
        self.tablaUsuarios = QtWidgets.QTableWidget(self.frame2)
        self.tablaUsuarios.setGeometry(QtCore.QRect(50, 20, 471, 181))
        self.tablaUsuarios.setStyleSheet("background-color: rgb(255, 255, 255);")
        self.tablaUsuarios.setSelectionMode(QtWidgets.QAbstractItemView.SingleSelection)
        self.tablaUsuarios.setSelectionBehavior(QtWidgets.QAbstractItemView.SelectRows)
        self.tablaUsuarios.setObjectName("tablaUsuarios")
        self.tablaUsuarios.setColumnCount(2)
        self.tablaUsuarios.setRowCount(0)
        item = QtWidgets.QTableWidgetItem()
        self.tablaUsuarios.setHorizontalHeaderItem(0, item)
        item = QtWidgets.QTableWidgetItem()
        self.tablaUsuarios.setHorizontalHeaderItem(1, item)
        self.frame3 = QtWidgets.QFrame(self.frame)
        self.frame3.setGeometry(QtCore.QRect(10, 0, 881, 491))
        self.frame3.setStyleSheet("background-color: rgb(255, 255, 255);")
        self.frame3.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.frame3.setFrameShadow(QtWidgets.QFrame.Raised)
        self.frame3.setObjectName("frame3")
        self.tableWidget = QtWidgets.QTableWidget(self.frame3)
        self.tableWidget.setGeometry(QtCore.QRect(20, 60, 851, 391))
        self.tableWidget.setSelectionMode(QtWidgets.QAbstractItemView.SingleSelection)
        self.tableWidget.setSelectionBehavior(QtWidgets.QAbstractItemView.SelectRows)
        self.tableWidget.setObjectName("tableWidget")
        self.tableWidget.setColumnCount(9)
        self.tableWidget.setRowCount(0)
        item = QtWidgets.QTableWidgetItem()
        self.tableWidget.setHorizontalHeaderItem(0, item)
        item = QtWidgets.QTableWidgetItem()
        self.tableWidget.setHorizontalHeaderItem(1, item)
        item = QtWidgets.QTableWidgetItem()
        self.tableWidget.setHorizontalHeaderItem(2, item)
        item = QtWidgets.QTableWidgetItem()
        self.tableWidget.setHorizontalHeaderItem(3, item)
        item = QtWidgets.QTableWidgetItem()
        self.tableWidget.setHorizontalHeaderItem(4, item)
        item = QtWidgets.QTableWidgetItem()
        self.tableWidget.setHorizontalHeaderItem(5, item)
        item = QtWidgets.QTableWidgetItem()
        self.tableWidget.setHorizontalHeaderItem(6, item)
        item = QtWidgets.QTableWidgetItem()
        self.tableWidget.setHorizontalHeaderItem(7, item)
        item = QtWidgets.QTableWidgetItem()
        self.tableWidget.setHorizontalHeaderItem(8, item)
        self.label_9 = QtWidgets.QLabel(self.centralwidget)
        self.label_9.setGeometry(QtCore.QRect(20, 20, 891, 41))
        font = QtGui.QFont()
        font.setFamily("Verdana")
        font.setPointSize(22)
        font.setBold(True)
        font.setWeight(75)
        self.label_9.setFont(font)
        self.label_9.setObjectName("label_9")
        self.btnEditar = QtWidgets.QPushButton(self.centralwidget)
        self.btnEditar.setGeometry(QtCore.QRect(520, 470, 161, 51))
        font = QtGui.QFont()
        font.setFamily("Verdana")
        font.setPointSize(16)
        font.setBold(True)
        font.setWeight(75)
        self.btnEditar.setFont(font)
        self.btnEditar.setStyleSheet("color: rgb(255, 255, 255);\n"
"background-color: rgb(0, 0, 0);\n"
"border-radius: 15px;")
        self.btnEditar.setObjectName("btnEditar")
        self.btnEditar.raise_()
        self.tablaVenci.raise_()
        self.btnAnadir.raise_()
        self.frame.raise_()
        self.label_9.raise_()
        MainWindow.setCentralWidget(self.centralwidget)

        self.retranslateUi(MainWindow)
        QtCore.QMetaObject.connectSlotsByName(MainWindow)

#-------------------------------------------------------------------------------#
        self.filaTabla = 0
        self.selectedId = 0
        self.selectedIdM = 0
        self.estado = "DEFAULT"

        self.frame.hide()
        self.frame2.hide()
        self.frame3.hide()
        self.txtNroU.hide()
        self.txtCerrado.hide()
        
        

#-------------------------------------------------------------------------------#
#BOTONES
        self.btnGuardar.clicked.connect(lambda: controlMain.nuevoRecordatorio(self))
        self.pushButton.clicked.connect(lambda: controlMain.userSelect(self))
        self.btnAnadir.clicked.connect(lambda: controlMain.mostrarFrame(self))
        self.btnCancelar.clicked.connect(lambda: controlMain.ocultarFrame(self))
        self.btnEditar.clicked.connect(lambda: controlMain.editSelected(self))
        self.tablaUsuarios.doubleClicked.connect(lambda: controlMain.doubleClickUser(self))
        self.tableWidget.doubleClicked.connect(lambda: controlMain.doubleClickEditar(self))
        controlMain.mostrarVencimientos(self)

#-------------------------------------------------------------------------------#

    def retranslateUi(self, MainWindow):
        _translate = QtCore.QCoreApplication.translate
        MainWindow.setWindowTitle(_translate("MainWindow", "Recordatorio de Vencimientos"))
        item = self.tablaVenci.horizontalHeaderItem(0)
        item.setText(_translate("MainWindow", "Fecha Carga"))
        item = self.tablaVenci.horizontalHeaderItem(1)
        item.setText(_translate("MainWindow", "Nro Tramite"))
        item = self.tablaVenci.horizontalHeaderItem(2)
        item.setText(_translate("MainWindow", "Observacion"))
        item = self.tablaVenci.horizontalHeaderItem(3)
        item.setText(_translate("MainWindow", "Vencimiento"))
        item = self.tablaVenci.horizontalHeaderItem(4)
        item.setText(_translate("MainWindow", "Detalle"))
        self.btnAnadir.setText(_translate("MainWindow", "Añadir"))
        self.label.setText(_translate("MainWindow", "Usuario"))
        self.dateCarga.setDisplayFormat(_translate("MainWindow", "yyyy/M/dd"))
        self.btnGuardar.setText(_translate("MainWindow", "Guardar"))
        self.btnCancelar.setText(_translate("MainWindow", "Cancelar"))
        self.pushButton.setText(_translate("MainWindow", "Buscar"))
        self.label_10.setText(_translate("MainWindow", "Fecha de Carga"))
        self.label_2.setText(_translate("MainWindow", "Recordar a X días"))
        self.label_3.setText(_translate("MainWindow", "Vencimiento"))
        self.label_6.setText(_translate("MainWindow", "Nro Tramite"))
        self.checkBox.setText(_translate("MainWindow", "Recordatorio Abrierto"))
        self.label_7.setText(_translate("MainWindow", "Observacion"))
        self.label_11.setText(_translate("MainWindow", "Detalle"))
        item = self.tablaUsuarios.horizontalHeaderItem(0)
        item.setText(_translate("MainWindow", "ID"))
        item = self.tablaUsuarios.horizontalHeaderItem(1)
        item.setText(_translate("MainWindow", "NOMBRE USUARIO"))
        item = self.tableWidget.horizontalHeaderItem(0)
        item.setText(_translate("MainWindow", "ID"))
        item = self.tableWidget.horizontalHeaderItem(1)
        item.setText(_translate("MainWindow", "F_ Carga"))
        item = self.tableWidget.horizontalHeaderItem(2)
        item.setText(_translate("MainWindow", "Dias"))
        item = self.tableWidget.horizontalHeaderItem(3)
        item.setText(_translate("MainWindow", "Vence"))
        item = self.tableWidget.horizontalHeaderItem(4)
        item.setText(_translate("MainWindow", "ID _U"))
        item = self.tableWidget.horizontalHeaderItem(5)
        item.setText(_translate("MainWindow", "Obs"))
        item = self.tableWidget.horizontalHeaderItem(6)
        item.setText(_translate("MainWindow", "Nro T"))
        item = self.tableWidget.horizontalHeaderItem(7)
        item.setText(_translate("MainWindow", "Cerrado"))
        item = self.tableWidget.horizontalHeaderItem(8)
        item.setText(_translate("MainWindow", "Detalle"))
        self.label_9.setText(_translate("MainWindow", "Recordatorios"))
        self.btnEditar.setText(_translate("MainWindow", "Editar"))


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    MainWindow = QtWidgets.QMainWindow()
    ui = Ui_MainWindow()
    ui.setupUi(MainWindow)
    MainWindow.show()
    sys.exit(app.exec_())
